test/puppetca_token_whitelisting/puppetca_token_whitelisting_autosigner_test.rb

```diff
@@ -0,0 +1,96 @@
+require 'test_helper'
+require 'tempfile'
+require 'fileutils'
+require 'openssl'
+require 'yaml'
+require 'json'
+
+require 'puppetca/puppetca'
+require 'puppetca_token_whitelisting/puppetca_token_whitelisting'
+require 'puppetca_token_whitelisting/puppetca_token_whitelisting_autosigner'
+require 'puppetca_token_whitelisting/puppetca_token_whitelisting_csr'
+
+class PuppetCaTokenWhitelistingAutosignerTest < Test::Unit::TestCase
+  def setup
+    @file = Tempfile.new('autosign_test')
+    begin
+      ## Setup
+      FileUtils.cp './test/fixtures/puppetca/hosts.yml', @file.path
+    rescue
+      @file.close
+      @file.unlink
+      @file = nil
+    end
+    @autosigner = Proxy::PuppetCa::TokenWhitelisting::Autosigner.new
+    @autosigner.stubs(:hosts_file).returns(@file.path)
+    rsa_cert = OpenSSL::PKey::RSA.generate 2048
```

Comment:
Does this mean a RSA cert is generated for every test in this file? Isn't that somewhat slow?

Reply:
Not noticeable (at least for me). But since we decided to change the implementation to save the tokens instead of certnames, we will need to supply a fixtured cert anyway.